perllib/FixMyStreet/Queue/Item/Report.pm

```diff
@@ -235,6 +262,8 @@ sub _send {
         }
     }
 
+    $self->report->add_send_fail_type($_) for @send_fail_types;
```

Comment:
There was a reason but I need to remember what it was.

Reply:
If doing it inside the loop and, say, you're looping through send methods (Open311, Email):
- if Open311 fails, it will be added to `send_fail_body_ids` [I changed to that from `send_fail_type`] - no problems there
- but then in the next iteration of the loop, we're looking at Email, and this block will cause sending to skip as we now have a populated send_fail_body_ids array but no 'Email' body in it:
`if ( @{ $report->send_fail_body_ids }
            && !$report->has_given_send_fail_body_id($body_id) )
        {
            $sender->success(1);
            next;
        }`